packages/bson-transpilers/codegeneration/CodeGenerationVisitor.js

```diff
@@ -24,6 +25,7 @@ module.exports = (ANTLRVisitor) => class CodeGenerationVisitor extends ANTLRVisi
     super();
     this.idiomatic = true; // PUBLIC
     this.clearImports();
+    this.state = { declarationStore: new DeclarationStore() };
```

Comment:
@alenakhineika @addaleax @mcasimir I ran into an issue concerning the way we handle the `argsTemplate` functions.  Since they are variadic, passing in the declaration store as the last value will cause a lot of problems with existing language templates.  [This one](https://github.com/mongodb-js/compass/blob/main/packages/bson-transpilers/symbols/csharp/templates.yaml#L770), in particular, comes to mind.  To fix this, I  propose

1. Adding this state hash in case we ever need to add state for something else in the future
2. Updating all uses of `argsTemplate` so that the first argument will be be `this.state`